elyra/metadata/schemas/code-snippet.json

```diff
@@ -0,0 +1,32 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Schema for Code Snippet metadata",
+  "name": "code-snippet",
+  "properties": {
+    "schema_name": {
+      "type": "string",
+      "pattern": "^[a-z][a-z0-9-_]*[a-z0-9]$",
+      "minLength": 1
+    },
+    "display_name": {
+      "description": "The display name of the metadata",
+      "type": "string",
+      "pattern": "^[a-zA-Z][a-zA-Z0-9-_. (){}]*[a-zA-Z0-9)}]$"
+    },
+    "metadata": {
+      "description": "Additional data specific to this metadata",
+      "type": "object",
+      "properties": {
+        "language": {
+          "description": "Code snippet implementation language",
+          "type": "string"
+        },
+        "code": {
+          "type": "array"
```

Comment:
Is there a reason for storing the code as an array? I understand the notebook stores data in its `"source":` elements as an array - one item per code line - but it seems more natural to store the code in a free-form string.

Is the widget simply responsible for placing the code into the clipboard, then the user pastes the clipboard into the appropriate cell of the notebook?  If so, I would think it would be better to treat the entry as a string rather an array.  A string is also easier to edit, etc. in case someone wants to edit a code-snippet outside the application.

Probably not a big deal, just curious.